contracts/src/v0.8/functions/tests/1_0_0/FunctionsCoordinator.t.sol

```diff
@@ -115,8 +121,54 @@ contract FunctionsBilling__GetJuelsPerGas {
 }
 
 /// @notice #estimateCost
-contract FunctionsBilling_EstimateCost {
-
+contract FunctionsBilling_EstimateCost is FunctionsSubscriptionSetup {
+  function setUp() public virtual override {
+    FunctionsSubscriptionSetup.setUp();
+
+    // Get cost estimate as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+  }
+
+  uint256 private constant REASONABLE_GAS_PRICE_CEILING = 1_000_000_000_000_000; // 1 million gwei
+
+  function test_EstimateCost_RevertsIfGasPriceAboveCeiling() public {
+    // Build minimal valid request data
+    string memory sourceCode = "return 'hello world';";
+    FunctionsRequest.Request memory request;
+    FunctionsRequest.initializeRequest(
+      request,
+      FunctionsRequest.Location.Inline,
+      FunctionsRequest.CodeLanguage.JavaScript,
+      sourceCode
+    );
+    bytes memory requestData = FunctionsRequest.encodeCBOR(request);
+
+    uint32 callbackGasLimit = 5_500;
+    uint256 gasPriceWei = REASONABLE_GAS_PRICE_CEILING + 1;
+
+    vm.expectRevert(FunctionsBilling.InvalidCalldata.selector);
+
+    s_functionsCoordinator.estimateCost(s_subscriptionId, requestData, callbackGasLimit, gasPriceWei);
+  }
+
+  function test_EstimateCost_Success() public view {
+    // Build minimal valid request data
+    string memory sourceCode = "return 'hello world';";
+    FunctionsRequest.Request memory request;
+    FunctionsRequest.initializeRequest(
+      request,
+      FunctionsRequest.Location.Inline,
+      FunctionsRequest.CodeLanguage.JavaScript,
+      sourceCode
+    );
+    bytes memory requestData = FunctionsRequest.encodeCBOR(request);
+
+    uint32 callbackGasLimit = 5_500;
+    uint256 gasPriceWei = 1;
+
+    s_functionsCoordinator.estimateCost(s_subscriptionId, requestData, callbackGasLimit, gasPriceWei);
```

Comment:
same here. Check returned value

Reply:
The second option is best, but IMO I would be fine with just getting the current value, making sure it makes sense and then hardcoding that as the expected value.

Reply:
Added a value snapshot